src/utils/temporal.cpp

```diff
@@ -514,6 +515,275 @@ size_t LocalDateTimeHash::operator()(const LocalDateTime &local_date_time) const
   return result;
 }
 
+namespace {
+inline constexpr auto *kSupportedZonedDateTimeFormatsHelpMessage = R"help(
+A string representing ZonedDateTime should have the following structure:
+
+- <DateTime><timezone>
+
+The <DateTime> substring should use one of the following formats:
+
+- YYYY-MM-DDThh:mm:ss
+- YYYY-MM-DDThh:mm
+
+or
+
+- YYYYMMDDThhmmss
+- YYYYMMDDThhmm
+- YYYYMMDDThh
+
+The <timezone> substring should use one of the following formats:
+
+- Z
+- ±hh:mm
+- ±hh:mm[ZoneName]
+- ±hhmm
+- ±hhmm[ZoneName]
+- ±hh
+- ±hh[ZoneName]
+- [ZoneName]
+
+Symbol table:
+|---|---------|
+| Y | YEAR    |
+|---|---------|
+| M | MONTH   |
+|---|---------|
+| D | DAY     |
+|---|---------|
+| h | HOURS   |
+|---|---------|
+| m | MINUTES |
+|---|---------|
+| s | SECONDS |
+|---|---------|
+| Z | UTC     |
+|---|---------|
+
+ZoneName is a standardized timezone name from the IANA time zone database, given without quote marks.
+
+Additionally, seconds can be defined as decimal fractions with 3 or 6 digits after the decimal point.
+First 3 digits represent milliseconds, while the second 3 digits represent microseconds.
+
+It's important to note that the date and time parts should use both the corresponding separators
+or both parts should be written in their basic forms without the separators.)help";
+}  // namespace
+
+Timezone ParseTimezoneFromName(std::string_view timezone_string) {
+  auto extract_timezone_name = [](std::string_view &string) {
+    if (string.front() != '[' or string.back() != ']') {
+      throw temporal::InvalidArgumentException("Timezone name is not enclosed by '[' ']'.");
+    }
+    string.remove_prefix(1);
+    string.remove_suffix(1);
+    return string;
+  };
+
+  const auto *timezone = [&]() {
+    try {
+      return std::chrono::locate_zone(extract_timezone_name(timezone_string));
+    } catch (const std::exception &_) {
+      throw temporal::InvalidArgumentException("Timezone name is not in the IANA time zone database.");
+    }
+    return (const std::chrono::time_zone *)nullptr;
+  }();
```

Comment:
Please prefer `std::invoke()` to legacy "immediately invoked function expressions".